Add output format for GitHub.

koozz opened the pull request:
GitHub output includes the specialized [workflow commands](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions) that will automatically be parsed by GitHub during a Pull Request and annotate the policy-violating file.

Example (using **koozz/conftest:v0.26.0-alpha** instead of **openpolicyagent/conftest** to be able to test this unmerged/unreleased code):

```yaml
---
name: Validate

on:
  pull_request:
    branches: 
      - main
    paths:
      - '**.yml'
      - '**.yaml'

jobs:
  validate:
    name: 'Rego'
    runs-on: ubuntu-latest
    container: koozz/conftest:v0.26.0-alpha
    steps:
      - name: Check out the code
        uses: actions/checkout@v2
      - name: Validate workflows
        run: |
          conftest test -o github .github/workflows/*.yml -p policy -n workflows
```

policy/workflows.rego:

```rego
package workflows

deny[msg] {
	not input.name
	msg = sprintf("Missing 'name' on: %s", [input])
}
```

The deny message can of course be as succinct and/or as pretty as you want, it will be outputted in the console and annotated on the file where the violation was.

GitHub action output:

```text
Run conftest test -o github .github/workflows/*.yml -p policy -n workflows
  conftest test -o github .github/workflows/*.yml -p policy -n workflows
  shell: sh -e {0}
Testing '.github/workflows/validate.yml' against 1 policies in namespace 'workflows'
  success file=.github/workflows/validate.yml 1
Testing '.github/workflows/yamllint.yml' against 1 policies in namespace 'workflows'
  Error: Missing 'name' on: {"jobs": {"linter": {"runs-on": "ubuntu-latest", "steps": [{"name": "Check out code", "uses": "actions/checkout@v2"}, {"name": "Run yamllint", "uses": "actionshub/yamllint@v1.0.0"}]}}, "true": {"push": {"branches-ignore": ["main"], "paths": ["**.yml", "**.yaml"]}}}
2 tests, 1 passed, 0 warnings, 1 failure, 0 exceptions
Error: Process completed with exit code 1.
```

Annotated file:

![image](https://user-images.githubusercontent.com/264372/123966366-a8d87c80-d9b5-11eb-83ad-763de75f93d2.png)

srenatus commented on output/github.go:
```diff
@@ -0,0 +1,94 @@
+// GitHub represents an Outputter that outputs
+// results in GitHub workflow format.
+type GitHub struct {
+	Writer io.Writer
```
🤔 I don't think we need to expose this, let's do
```suggestion
	writer io.Writer
```

...but it looks like this is consistent with the other plugins, so fine -- it's not a big deal anyways.

koozz replied:
I started off just as the other output formats, but you're right, io.Writer can be private.

srenatus commented on output/github.go:
```diff
@@ -0,0 +1,94 @@
+		totalFailures, pluralSuffixFailures,
+		totalExceptions, pluralSuffixExceptions,
+	)
+	fmt.Fprintf(t.Writer, "%s\n", outputText)
```
[nit]
```suggestion
	fmt.Fprintln(t.Writer, outputText)
```

srenatus reviewed (approved):
Code looks good to me (nitpicks only), but I haven't tested it myself 😃

koozz commented:
Sorry for the updates, I accepted a code review suggestion, but that wasn't signed-off.

mykter commented on output/github.go:
```diff
@@ -0,0 +1,94 @@
+		}
+
+		if result.Successes > 0 {
+			fmt.Fprintf(t.writer, "success file=%v %v\n", result.FileName, result.Successes)
```
is this meant to not have the :: like the others?

mykter replied:
worth adding a test with a success case in perhaps

koozz replied:
Sadly only errors and warning are annotated in GitHub PR's and no success is set. If there are no errors, the check will return 'green'. The '::' is something that GitHub picks up.

koozz replied:
I have removed all the double colons where the output is not GitHub flavored (exception/skipped/success).

mykter commented on output/github.go:
```diff
@@ -0,0 +1,94 @@
+)
+
+// GitHub represents an Outputter that outputs
+// results in GitHub workflow format.
```
a link to the spec would be useful here

koozz replied:
Added a link to the GitHub documentation on workflow commands.

koozz commented:
@mykter I've updated the code since your review and added a link to the documentation. Is there anything else I can do? Do I need to ask a specific person for final approval?

koozz commented:
@srenatus and @mykter I have just squashed the 'fix'-commits.

jalseth reviewed (approved):
This looks good to me! Thank you for the contribution.
